src/streams.ts

```diff
@@ -42,3 +43,36 @@ export function compose(streams: NodeJS.ReadWriteStream[]) {
     return new PassThrough({objectMode: true});
   }
 }
+
+/**
+ * A stream that passes through Vinyl file objects and loads any files
+ * that have not yet had their content loaded.
```

Comment:
after our talk, turn this into a VinylReaderTransform and make the StreamAnalyzer's dependencies stream a stream of filenames.